todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModel.java

```diff
@@ -16,39 +16,69 @@
 
 package com.example.android.architecture.blueprints.todoapp.tasks;
 
-import static com.example.android.architecture.blueprints.todoapp.tasks.TasksFilterType.ALL_TASKS;
-
 import android.content.Context;
 import android.databinding.BaseObservable;
 import android.databinding.Bindable;
+import android.databinding.ObservableArrayList;
+import android.databinding.ObservableBoolean;
+import android.databinding.ObservableField;
+import android.databinding.ObservableList;
 import android.graphics.drawable.Drawable;
 
 import com.example.android.architecture.blueprints.todoapp.BR;
 import com.example.android.architecture.blueprints.todoapp.R;
+import com.example.android.architecture.blueprints.todoapp.SnackBarChangedCallback;
+import com.example.android.architecture.blueprints.todoapp.data.Task;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+import com.example.android.architecture.blueprints.todoapp.util.EspressoIdlingResource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.example.android.architecture.blueprints.todoapp.tasks.TasksFilterType.ALL_TASKS;
 
 /**
- * Exposes the data to be used in the {@link TasksContract.View}.
+ * Exposes the data to be used in the task list screen.
  * <p>
  * {@link BaseObservable} implements a listener registration mechanism which is notified when a
  * property changes. This is done by assigning a {@link Bindable} annotation to the property's
  * getter method.
  */
-public class TasksViewModel extends BaseObservable {
+public class TasksViewModel extends BaseObservable implements SnackBarChangedCallback.SnackBarViewModel {
+
+    private final TasksRepository mTasksRepository;
+
+    private final TasksNavigator mNavigator;
+
+    private TasksFilterType mCurrentFiltering = TasksFilterType.ALL_TASKS;
+
+    public final ObservableList<Task> items = new ObservableArrayList<>();
 
-    int mTaskListSize = 0;
+    public final ObservableBoolean dataLoading = new ObservableBoolean(false);
 
-    private final TasksContract.Presenter mPresenter;
+    private final ObservableBoolean isDataLoadingError = new ObservableBoolean(false);
```

Comment:
The [naming conventions](https://source.android.com/source/code-style.html#follow-field-naming-conventions) say that non-public field names start with "m".
So this means that we should have `mIsDataLoadingError` and `mSnackBarText`